Pod/Classes/SEGMoEngageIntegration.m

```diff
@@ -25,47 +25,79 @@ - (void)track:(SEGTrackPayload *)payload
 
 - (void)identify:(SEGIdentifyPayload *)payload
 {
-    NSDictionary *traits = payload.traits;
-
+    NSDictionary *moengagePayloadDict = [payload.traits copy];
+    
+    NSMutableDictionary *traits = [NSMutableDictionary dictionaryWithDictionary:moengagePayloadDict];
+    if(![traits.allKeys count]){
```

Comment:
If the dictionary is empty, `traits.count` would be zero which would be falsy and not enter this block.